deployment/environment.go

```diff
@@ -238,17 +243,50 @@ func MustPeerIDFromString(s string) p2pkey.PeerID {
 }
 
 type NodeChainConfigsLister interface {
+	ListNodes(ctx context.Context, in *nodev1.ListNodesRequest, opts ...grpc.CallOption) (*nodev1.ListNodesResponse, error)
 	ListNodeChainConfigs(ctx context.Context, in *nodev1.ListNodeChainConfigsRequest, opts ...grpc.CallOption) (*nodev1.ListNodeChainConfigsResponse, error)
 }
 
 // Gathers all the node info through JD required to be able to set
-// OCR config for example.
+// OCR config for example. nodeIDs can be JD IDs or PeerIDs
 func NodeInfo(nodeIDs []string, oc NodeChainConfigsLister) (Nodes, error) {
+	if len(nodeIDs) == 0 {
+		return nil, nil
+	}
+	// if nodeIDs starts with `p2p_` lookup by p2p_id instead
+	filterByPeerIDs := strings.HasPrefix("p2p_", nodeIDs[0])
```

Comment:
A bit implicit, but JD node IDs start with `node_`. `p2p_` is a good indicator to filter by peer ID instead